variables.tf

```diff
@@ -1,11 +1,252 @@
+  validation {
+    error_message = "Invalid value for `notification_events`."
+
+    condition = (
+      length(var.scan_notification_configs) == 0 ||
+      toset(flatten(
+        [
+          for config in var.scan_notification_configs : config.notification_events
+        ]
+        )) == setintersection(
+        [
+          "All",
+          "InProgress",
+          "Success",
+          "TimedOut",
+          "Cancelled",
+          "Failed"
+        ],
+        toset(flatten(
+          [
+            for config in var.scan_notification_configs : config.notification_events
+          ]
+        ))
+      )
+    )
+  }
+
+  validation {
+    error_message = "Invalid `notification_type`."
+
+    condition = (
+      length(var.scan_notification_configs) == 0 ||
+      toset(
+        [for config in var.scan_notification_configs : config.notification_type]
+        ) == setintersection(
+        [for config in var.scan_notification_configs : config.notification_type],
+        [
+          "Command",
+          "Invocation"
+        ]
+      )
+    )
+  }
+}
+
+variable "scan_schedule" {
+  description = "6-field Cron expression describing the scan maintenance schedule"
+  type        = string
+}
+
+variable "schedule_timezone" {
+  type        = string
+  description = "IANA format timezone to use for Maintenance Window scheduling"
+  default     = "UTC"
+
+  validation {
+    condition     = fileexists("/usr/share/zoneinfo.default/${var.schedule_timezone}") || fileexists("/usr/share/zoneinfo/${var.schedule_timezone}")
+    error_message = "Time zone invalid or validation failed."
+  }
+}
+
+variable "install_cutoff" {
+  default     = 1
+  description = "How many hours before the end of the maintenance Window to stop scheduling new instances to install patches"
+  type        = number
+}
+
+variable "install_duration" {
+  default     = 3
+  description = "How long in hours for the install maintenance window"
+  type        = number
+}
+
+variable "install_log_prefix" {
+  default     = "/patch_manager/install/"
+  description = "The S3 bucket subfolder to store install logs in"
+  type        = string
+}
+
+variable "install_notification_configs" {
+  default     = []
+  description = "A set of objects containing `notification_config`s [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_maintenance_window_task#notification_config)"
+
+  type = set(object({
+    notification_arn    = string
+    notification_events = string
+    notification_type   = string
+  }))
+
+  validation {
+    error_message = "Invalid value for `notification_events`."
+
+    condition = (
+      length(var.install_notification_configs) == 0 ||
+      toset(flatten(
+        [
+          for config in var.install_notification_configs : config.notification_events
+        ]
+        )) == setintersection(
+        [
+          "All",
+          "InProgress",
+          "Success",
+          "TimedOut",
+          "Cancelled",
```

Comment:
OMG is it misspelled in the TF docs or the API? https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_maintenance_window_task#notification_events

Reply:
It's not a misspelling, `misspell` is being too opinionated and the flags to ignore this aren't working. `Cancelled` and `canceled` are both correct, with the latter being more common in American english.

AWS does spell it `Cancelled` in this part of the API